sources/filesystem/readers.py

```diff
@@ -74,6 +84,60 @@ def _read_parquet(
                 yield rows.to_pylist()
 
 
+def _read_csv_duckdb(
+    items: Iterator[FileItemDict],
+    bucket: str,
+    chunk_size: Optional[int] = 5000,
+    use_pyarrow: bool = False,
+    credentials: Union[FileSystemCredentials, AbstractFileSystem] = dlt.secrets.value,
+    read_csv_kwargs: Optional[Dict[str, Any]] = {},
+) -> Iterator[TDataItems]:
+    """A resource to extract data from the given CSV files.
+
+    Uses DuckDB engine to import and cast CSV data.
+
+    Args:
+        items (Iterator[FileItemDict]): CSV files to read.
+        bucket (str): The bucket name.
+        chunk_size (Optional[int]):
+            The number of rows to read at once. Defaults to 5000.
+        use_pyarrow (bool):
+            Whether to use `pyarrow` to read the data and designate
+            data schema. If set to False (by default), JSON is used.
+        credentials (Union[FileSystemCredentials, AbstractFileSystem]):
+            The credentials to use. Defaults to dlt.secrets.value.
+        read_csv_kwargs (Optional[Dict]):
+            Additional keyword arguments to pass to the `read_csv()`.
+
+    Returns:
+        Iterable[TDataItem]: Data items, read from the given CSV files.
+    """
+    import duckdb
+    import fsspec  # type: ignore
+    import pendulum
+
+    config = FilesystemConfiguration(bucket, credentials)
```

Comment:
you do not need that. FileItemDict has methods to (1) open referenced file (2) get pre-authorized filesystem, look what existing csv reader does:
```
    for file_obj in items:
        # Here we use pandas chunksize to read the file in chunks and avoid loading the whole file
        # in memory.
        with file_obj.open() as file:
            for df in pd.read_csv(file, **kwargs):
                yield df.to_dict(orient="records")
```
user `open()` below. it will be already authenticated. (and maybe we do not need this `dlt` extension)

Reply:
A-ah, okay. When `fsspec` rejected a `FileItemDict` object with the comment "not a file-like" I didn't check that it actually openable, not just a dict. Done.